tgui/packages/tgui/interfaces/common/AtmosHandbook.tsx

```diff
@@ -0,0 +1,242 @@
+  name: string;
+  description: string;
+  factors: Factor[];
+};
+
+type Gas = {
+  id: string;
+  name: string;
+  description: string;
+  specific_heat: number;
+  reactions: { [key: string]: string } | [];
+};
+
+class ChangingInput extends Component<
+  {
+    title: InfernoNode | string;
+    icon: string;
+    onChange: (inputValue: string) => void;
+  },
+  { activeInput: boolean }
+> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeInput: false,
+    };
+  }
+
+  modifiedOnChange(value) {
+    this.setState({ activeInput: false });
+    this.props.onChange(value);
+  }
+
+  toggleInput() {
+    this.setState((prevState) => ({
+      activeInput: !prevState.activeInput,
+    }));
+  }
+
+  render() {
+    const { title, icon } = this.props;
+    return (
+      <Flex align="center">
+        <Flex.Item grow>
+          {this.state?.activeInput ? (
+            <Input
+              fluid
+              onChange={(e, value) => this.modifiedOnChange(value)}
+            />
+          ) : (
+            title
+          )}
+        </Flex.Item>
+        <Flex.Item>
+          <Button icon={icon} onClick={() => this.toggleInput()} />
+        </Flex.Item>
+      </Flex>
+    );
+  }
+}
+
+const GasHandbook = (props: { gasTextBook: Gas[] }, context) => {
+  const { act, data } = useBackend(context);
+  const { gasTextBook } = props;
+  const [activeGasId, setActiveGasId] = useSharedState(
+    context,
+    'activeGasId',
+    ''
+  );
+  const [activeReactionId, setActiveReactionId] = useSharedState(
+    context,
+    'activeReactionId',
+    ''
+  );
+  const relevantGas = gasTextBook.find((gas) => gas.id === activeGasId);
+  return (
+    <Section
+      title={
+        <ChangingInput
+          title={relevantGas ? 'Gas: ' + relevantGas.name : 'Gas Lookup'}
+          icon="search"
+          onChange={(keyword) =>
+            setActiveGasId(
+              gasTextBook.find((gas) =>
+                gas.name.toLowerCase().startsWith(keyword.toLocaleLowerCase())
+              )?.id || '')}
+        />
+      }
+      fill
+      scrollable>
+      {relevantGas && (
+        <>
+          <Box mb={'0.5em'}>{relevantGas.description}</Box>
+          <Box mb={'0.5em'}>
+            {'Specific heat: ' + relevantGas.specific_heat + ' Joule/KelvinMol'}
+          </Box>
+          <Box mb={'0.5em'}>{'Relevant Reactions:'}</Box>
+          {Object.keys(relevantGas.reactions).map((reaction_id) => (
+            <Flex key={reaction_id} mb="0.5em">
+              <Flex.Item
+                className="AtmosHandbook__link-content"
+                shrink
+                onClick={() => setActiveReactionId(reaction_id)}>
+                {relevantGas.reactions[reaction_id]}
+              </Flex.Item>
+            </Flex>
+          ))}
+        </>
+      )}
+    </Section>
+  );
+};
+
+const ReactionHandbook = (props: { reactionTextBook: Reaction[] }, context) => {
+  const { act, data } = useBackend(context);
```

Comment:
Since you're using TypeScript, you should either not include data, or type it with `useBackend<T>(context)`.